pkg/types/dsse/dsse.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2025 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dsse
+
+import (
+	"fmt"
+
+	pb "github.com/sigstore/rekor-tiles/pkg/generated/protobuf"
+)
+
+func Validate(ds *pb.DSSERequest) error {
+	if ds.Envelope == "" {
+		return fmt.Errorf("missing envelope")
+	}
+	if len(ds.Verifier) == 0 {
```

Comment:
Can we also add a function to validate verifiers contain non-empty `raw_bytes` and an identifier for a public key, or non-empty `raw_bytes` for a cert?

Reply:
Ah I see you started doing that in hashedrekord.go, we could extract that into a func to validate verifiers in both types.

Reply:
Moved to pkg/types/verifier